File: front/src/types/stamp.ts
import { StampAttendRestStatus, StampStatus } from '@/enums/stamp'

export type StampList = {
  attend_date: string
  leaving_date: string
  rest_date: string
  working_date: string
}

export type StampsByEmployeeId = {
  id: number
  employee_id: number
  status: StampAttendRestStatus
  stamp_start_date: string
  stamp_start: string
  stamp_end_date?: string
  stamp_end?: string
}

export type StampDetail = {
  id: number
  status: StampStatus
  stamp_start_date: string
  stamp_end_date?: string
}

export type CreateStampData = {
  employeeId: number
  status: StampStatus
  today: string
}

export type UpdateStampData = {
  id: number
  status: StampAttendRestStatus
  stamp_start_date: Date|null
  stamp_end_date: Date|null
}

export type AdminCreateStampData = {
  employeeId: number
  status: StampAttendRestStatus
  stamp_start_date: Date
  stamp_end_date: Date
}

File: api/entity/stamp/stamp.go
/*
エンティティ（出退勤）
*/
package stamp

type ID uint
type EmployeeID uint
type Status uint
type StampStartDate string
type StampStart string
type StampEndDate string
type StampEnd string

type Stamp struct {
	ID             ID             `json:"id"`
	EmployeeID     EmployeeID     `json:"employee_id"`
	Status         Status         `json:"status"`
	StampStartDate StampStartDate `json:"stamp_start_date"`
	StampStart     StampStart     `json:"stamp_start"`
	StampEndDate   StampEndDate   `json:"stamp_end_date"`
	StampEnd       StampEnd       `json:"stamp_end"`
}

type Stamps []Stamp
